py4j-java/src/py4j/commands/Command.java

```diff
@@ -88,9 +88,10 @@ public void execute(String commandName, BufferedReader reader,
 	 * <p>
 	 * Called when a command instance is created and assigned to a connection.
 	 * </p>
-	 * 
+	 *
 	 * @param gateway
+	 * @param connection the {@link GatewayConnection} this socket is assigned to
 	 */
-	public void init(Gateway gateway);
+	public void init(Gateway gateway, GatewayConnection connection);
```

Comment:
please ignore that comment. it was a bit too soon in the morning when I read the code.

You can keep it there, but before making the next release, I'll probably remove the init method altogether and just pass the Gateway and GatewayConnection instances to the execute() method. Since we're breaking backward compatibility, let's break it for a good cause :-)

Unless I'm mistaken, if we send the GatewayCOnnection instance to the execute method, all commands would not need to keep any state at all, which would definitively improve the design.